compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpInterceptorServiceValidator.java

```diff
@@ -61,36 +67,53 @@ public void perform(SyntaxNodeAnalysisContext syntaxNodeAnalysisContext) {
         validateInterceptorTypeAndProceed(syntaxNodeAnalysisContext, members);
     }
 
-    private static void validateInterceptorTypeAndProceed(SyntaxNodeAnalysisContext ctx, NodeList<Node> members) {
-        String interceptorType = null;
+    private void validateInterceptorTypeAndProceed(SyntaxNodeAnalysisContext ctx, NodeList<Node> members) {
         for (Node member : members) {
             if (member.kind() == SyntaxKind.TYPE_REFERENCE) {
-                String typeReference = ((TypeReferenceNode) member).typeName().toString();
-                switch (typeReference) {
-                    case Constants.HTTP_REQUEST_INTERCEPTOR:
-                        if (interceptorType == null) {
-                            interceptorType = Constants.REQUEST_INTERCEPTOR;
-                            break;
-                        } else {
-                            reportMultipleReferencesFound(ctx, (TypeReferenceNode) member);
-                            return;
-                        }
-                    case Constants.HTTP_REQUEST_ERROR_INTERCEPTOR:
-                        if (interceptorType == null) {
-                            interceptorType = Constants.REQUEST_ERROR_INTERCEPTOR;
-                            break;
-                        } else {
-                            reportMultipleReferencesFound(ctx, (TypeReferenceNode) member);
-                            return;
-                        }
+                Optional<Symbol> optionalTypeReferenceSymbol = ctx.semanticModel().symbol(member);
+                if (optionalTypeReferenceSymbol.isPresent()) {
+                    TypeReferenceTypeSymbol typeReferenceSymbol = (TypeReferenceTypeSymbol)
+                            optionalTypeReferenceSymbol.get();
+                    populateTypeReferenceName(ctx, member, typeReferenceSymbol);
                 }
             }
         }
         if (interceptorType != null) {
```

Comment:
For null-checks lets use java util methods.
```suggestion
        if (Objects.nonNull(interceptorType)) {
```

Reply:
Fixed in [`46596b5 `](https://github.com/ballerina-platform/module-ballerina-http/pull/739/commits/46596b5bf7c18942201cf955f8b0ee0aa85776e2)